### PR TITLE
docs: Add multi-lang module structure page

### PR DESCRIPTION
**Pull request by vikram-dagger**

### CODE REVIEW

**Comment by vikram-dagger** [comment]
@helderco I've created this as a separate multi-language page but on review, it seems like this is specific enough to each SDK that it should actually be part of the language-specific page (#7408). Let me know if you agree.

**Comment by vikram-dagger** [comment]
This was a difficult page to merge, because the content was significantly different for each languages. I needed to backfill some content and also adjust the structure of the sections. To confirm there are no errors, I would like review and approval from experts in our team. Pinging:

- @vito @sipsma @jedevc for Go
- @helderco for Python
- @TomChv for TypeScript

When reviewing, if you see areas where we can modify or optimize the internal page structure further, please add comments.

File: docs/current_docs/manuals/developer/module-structure.mdx [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+```toml
+[tool.dagger]
+use-uv = false
+```
+
+Note, however, that [pinning](./python/077133-python-dependencies.mdx#lock-file) via `requirements.lock` is only supported with `uv`.
+</TabItem>
+<TabItem value="TypeScript">
+
+The runtime container is currently hardcoded to run in Node.js 21.3 (although this may be configurable in future).
```
Comment (TomChv) [inline docs/current_docs/manuals/developer/module-structure.mdx:86]: This will change soon when we will use LTS version, see #7433

**Review by TomChv — COMMENTED** [review]
Thanks for these changes, I added some update suggestions based on new changes that were introduced with #7096.

I like the overall structure, you found a nice balance between generic information and language specific.

File: docs/current_docs/manuals/developer/module-structure.mdx [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+
+```json
+{
+  "dependencies": {
+    "typescript": "^5.3.2"
+  },
+  "devDependencies": {
+    "@dagger.io/dagger": "./sdk"
+  }
+}
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/module-structure.mdx:109]: Added

File: docs/current_docs/manuals/developer/module-structure.mdx [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+This is the most important requirement. The other conventions are flexible, as long as the above is verified.
+
+</TabItem>
+<TabItem value="TypeScript">
+Dagger modules in Typescript are built to be installed, like libraries. The runtime container installs the module code with:
+
+```shell
+# executed by the runtime container
+npm install
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/module-structure.mdx:449]: Added

File: docs/current_docs/manuals/developer/module-structure.mdx [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+
+The runtime container is currently hardcoded to run in Node.js 21.3 (although this may be configurable in future).
+
+[Bun](https://bun.sh/) is experimentally supported and [work is in progress](https://github.com/dagger/dagger/issues/4368) to support [Deno](https://deno.com/).
+
+The TypeScript SDK is installed automatically, including dependencies, with a version that's tied to the currently running Dagger Engine container:
+
+```shell
+# executed by the runtime container
+npm install --package-lock-only ./sdk
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/module-structure.mdx:94]: Added, please check if correct

**Comment by vikram-dagger** [comment]
> Thanks for these changes, I added some update suggestions based on new changes that were introduced with #7096.

I've added these changes, can you please review once again @TomChv ? Thanks!

**Comment by vikram-dagger** [comment]
Ping @vito @sipsma @jedevc @helderco @TomChv for review, see https://github.com/dagger/dagger/pull/7411#issuecomment-2122665088

**Review by kpenfound — APPROVED** [review]
One change in comment

**Pull request merged**